mcr.cpp

```diff
@@ -1,59 +1,98 @@
-        win = true;
-    if (game[1][0] == game[1][1] && game[1][1] == game[1][2] && (game[1][0] == 'X' || game[1][0] == 'O'))
-        win = true;
-    if (game[2][0] == game[2][1] && game[2][1] == game[2][2] && (game[2][0] == 'X' || game[2][0] == 'O'))
-        win = true;
-    // column
-    if (game[0][0] == game[1][0] && game[1][0] == game[2][0] && (game[0][0] == 'X' || game[0][0] == 'O'))
-        win = true;
-    if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] == 'X' || game[0][1] == 'O'))
-        win = true;
-    if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] == 'X' || game[0][2] == 'O'))
-        win = true;
-    // diagonal
-    if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O'))
-        win = true;
-    if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O'))
-        win = true;
-    return win;
+    // Check rows
+    for (int i = 0; i < 3; i++)
+    {
+        if (game[i][0] == game[i][1] && game[i][1] == game[i][2] && (game[i][0] == 'X' || game[i][0] == 'O'))
+            return true;
+    }
+
+    // Check columns
+    for (int j = 0; j < 3; j++)
+    {
+        if (game[0][j] == game[1][j] && game[1][j] == game[2][j] && (game[0][j] == 'X' || game[0][j] == 'O'))
+            return true;
+    }
+
+    // Check diagonals
+    if ((game[0][0] == game[1][1] && game[1][1] == game[2][2]) || (game[0][2] == game[1][1] && game[1][1] == game[2][0]))
+    {
+        if (game[1][1] == 'X' || game[1][1] == 'O')
+            return true;
+    }
+
+    return false;
+}
+
+void printGame(char game[3][3])
+{
+    cout << "Game Now:" << endl;
+    cout << game[0][0] << " | " << game[0][1] << " | " << game[0][2] << endl;
+    cout << "---------" << endl;
+    cout << game[1][0] << " | " << game[1][1] << " | " << game[1][2] << endl;
+    cout << "---------" << endl;
+    cout << game[2][0] << " | " << game[2][1] << " | " << game[2][2] << endl;
 }
+
 int main()
 {
-    int i, j;
-    char game[3][3] = {' '}; // Tic-tac-toe
+    int i, j, n;
+    char game[3][3]; // Tic-tac-toe
+    memset(game, ' ', sizeof(game)); // initialize 3x3 table
     char player1 = 'X';
     char player2 = 'O';
     bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first.
+    cout<< "Welcome to Tic-Tac-Toe Game!" << endl;
     cout << "X = Player 1" << endl
          << "O = Player 2" << endl;
-    for (int n = 0; n < 9; n++)
+    for (n = 0; n < 9; n++)
     {
         turn = !turn; // use the not-operator to change true to false or false to true.
         if (turn == false)
             cout << "Player 1: ";
         else
             cout << "Player 2: ";
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
+        cout << "Which cell to mark? i:[0..2], j:[0..2]: ";
         cin >> i >> j;
```

Comment:
Without the considering the situation of illegal input, a criterion should be added to avoid input outside the chessboard.
Methods: Add while(!(i == 0||i == 1||i == 2) && !(j == 0 || j == 1 || j == 2)) { cin >> i >> j; } after cin >> i >> j;